config/config.exs

```diff
@@ -0,0 +1,67 @@
+# This file is responsible for configuring your application
+# and its dependencies with the aid of the Mix.Config module.
+#
+# This configuration file is loaded before any dependency and
+# is restricted to this project.
+use Mix.Config
+
+# Customize the firmware. Uncomment all or parts of the following
+# to add files to the root filesystem or modify the firmware
+# archive.
+
+# config :nerves, :firmware,
+#   rootfs_additions: "config/rootfs_additions",
+#   fwup_conf: "config/fwup.conf"
+
+# Import target specific config. This must remain at the bottom
+# of this file so it overrides the configuration defined above.
+# Uncomment to use target specific configurations
+
+# import_config "#{Mix.Project.config[:target]}.exs"
+
+network_interface = System.get_env("NERVES_NETWORK_INTERFACE") || "usb0"
+key_mgmt = System.get_env("NERVES_NETWORK_KEY_MGMT") || "WPA-PSK"
+test_server = System.get_env("NERVES_TEST_SERVER")
+websocket_protocol = System.get_env("WEBSOCKET_PROTOCOL") || "ws"
+
+config :bootloader,
+  app: :nerves_system_test,
+  init: [:nerves_runtime, :nerves_network]
+
+config :nerves_network, :default,
+  eth0: [
+    ipv4_address_method: :dhcp
+  ],
+  usb0: [
+    ipv4_address_method: :linklocal
+  ]
+if ssid = System.get_env("NERVES_NETWORK_SSID") do
+  config :nerves_network, :default,
+    wlan0: [
+      ssid: ssid,
+      psk: System.get_env("NERVES_NETWORK_PSK"),
+      key_mgmt: String.to_atom(key_mgmt)
+    ]
+end
+
+config :nerves_system_test, NervesTestServer.Socket,
+  url: "#{websocket_protocol}://#{test_server}/socket/websocket"
+
+config :nerves_system_test,
+  system: Mix.Project.config[:app],
+  network_interface: network_interface,
+  tests: [
+    {:nerves_system_test, :priv_dir, "test"}
+  ]
+
+config :nerves_runtime, :kv,
+  nerves_fw_application_part0_devpath: "/dev/mmcblk0p3",
```

Comment:
It would be great if these variables could be deleted. It looks like they're copy/paste from the defaults in the `fwup.conf` file and it would be better to not have to duplicate them. I don't think that I'd even want some of them duplicated, since it defeats the purpose of code introspecting the firmware to figure these things out.